hist/hist/src/TH2.cxx

```diff
@@ -701,6 +701,12 @@ void TH2::DoFitSlices(bool onX,
    if (lastbin < 0 || lastbin > nbins + 1) lastbin = nbins + 1;
    if (lastbin < firstbin) {firstbin = 0; lastbin = nbins + 1;}
    TString opt = option;
+   TString gcut("");
+   Int_t i1 = opt.Index("[");
+   if (i1>=0) {
+      Int_t i2 = opt.Index("]");
+      gcut = opt(i1,i2-i1+1);
+   }
```

Comment:
Simple code, but one should take into account several points.
First of all, avoid problem when syntax has mistake like when opt="[name" - no closing braces.
Second, one should suppress cutg name from further options decoding.
Third, syntax `"e" + gcut` is not safe.
I propose following code here:
``` 
   TString proj_opt = "e";
   Int_t i1 = opt.Index("[");
   Int_t i2 = opt.Index("]");
   if ((i1>=0) && (i2>i1)) {
      proj_opt += opt(i1,i2-i1+1);
      opt.Remove(i1, i2-i1+1);
   }
```
And of course, use proj_opt when calling Projection methods